### PR TITLE
Correct KeyError from matplotlib when processing Series yerr

### LINKED ISSUE #11858: Index without 0 in xerr/yerr causes KeyError
When creating an errorbar plot, if the data object being used for the errors (`xerr` or `yerr`) doesn't have 0 in its index, it produces a `KeyError` from matplotlib code. I can produce this with the following code sample:

```python
import pandas as pd, numpy as np
i = np.array([1,2,3])
a = pd.DataFrame(i, index=i)
a.plot(yerr=a)
```

The underlying bug (if it is a bug) in matplotlib is responsible for several other issues, including #4493 and #6127, but this case is different because it uses only Pandas API methods, rather than passing a Pandas object to a matplotlib method. So it's a little harder to justify passing this off on matplotlib to fix, as was done in e.g. #6127.

If the "proper" fix ever is implemented in matplotlib code, it should solve this as well as #4493 and #6127 and all the others of that nature, but until that point, Pandas can work around it by converting the error object (if it is a `Series` or `DataFrame`) to an `ndarray`. I'm working on this in 4b04f80c41684ba9ab05ce8c87b17b96fde87290 but I'm not sure if there's a better way to fix it, or if this breaks something. (If so, the tests don't indicate it.)

### PR DESCRIPTION
**Pull request by gliptak**
- [x] closes #11858
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tseries/tests/test_plotting.py [hunk 1]
```diff
@@ -76,6 +76,12 @@ def test_frame_inferred(self):
         df = DataFrame(np.random.randn(len(idx), 3), index=idx)
         _check_plot_works(df.plot)
 
+    def test_is_error_nozeroindex(self):
+        # GH11858
+        i = np.array([1, 2, 3])
+        a = DataFrame(i, index=i)
+        a.plot(yerr=a)
```
Comment (jreback) [inline pandas/tseries/tests/test_plotting.py]: use check_plot_works

Comment (gliptak) [inline pandas/tseries/tests/test_plotting.py]: updated

File: pandas/tools/plotting.py [hunk 2]
```diff
@@ -1331,6 +1331,9 @@ def _plot(cls, ax, x, y, style=None, is_errorbar=False, **kwds):
             x = x._mpl_repr()
 
         if is_errorbar:
+            if 'yerr' in kwds:
```
Comment (jreback) [inline pandas/tools/plotting.py]: ``kwds['yerr'] = com._values_from_object(kwds.get('yerr'))``

Comment (gliptak) [inline pandas/tools/plotting.py]: Changed

Comment (jreback) [inline pandas/tools/plotting.py]: is this also a problem with ``xerr``?

**Comment by jreback** [comment]
pls add a whatsnew note

**Comment by gliptak** [comment]
@jreback Which section would this go in whatsnew?

**Comment by jreback** [comment]
bug fix

File: pandas/tools/plotting.py (hunk 2 above)
Comment (gliptak) [inline pandas/tools/plotting.py]: adding step for xerr too

**Comment by tacaswell** [comment]
left note on orginal issue, but can you please check if this is still a problem on mpl 1.5.1, 2.x or master?  If so, please make an issue upstream.

**Comment by gliptak** [comment]
whatsnew added

**Comment by tacaswell** [comment]
Also see https://github.com/matplotlib/matplotlib/pull/6392

**Comment by gliptak** [comment]
@tacaswell Based on the assessment by @diazona  in https://github.com/pydata/pandas/pull/13114#issuecomment-218006317 , sending ```ndarray``` into ```matplotlib``` seems like a good modification (in addition to https://github.com/matplotlib/matplotlib/pull/6392 proposed by you)

**Comment by tacaswell** [comment]
That fixes the pandas problem, but not the case where

```python
ax.errorbar('x', 'y', yerr='ye', data=df)
```

This should probably go into pandas for use with non-patched versions of mpl, but in the long run mpl should be able to cope with `Series` anyplace we can cope with ndarray.

File: pandas/tools/plotting.py [hunk 3]
```diff
@@ -1331,6 +1331,10 @@ def _plot(cls, ax, x, y, style=None, is_errorbar=False, **kwds):
             x = x._mpl_repr()
 
         if is_errorbar:
+            if 'xerr' in kwds:
+                kwds['xerr'] = com._values_from_object(kwds.get('xerr'))
```
Comment (jreback) [inline pandas/tools/plotting.py]: actually just do:  ``np.array(kwds.get('xerr'))``

**Comment by diazona** [comment]
Yep, I've confirmed that a test program like the following still reproduces the `KeyError`. (The fix I was working on in diazona/pandas@4b04f80c41684ba9ab05ce8c87b17b96fde87290 doesn't help with this case either.) Maybe there's one single place to make a change that would cover all these cases?

    import matplotlib.pyplot as plt
    import pandas as pd
    import numpy as np

    df = pd.DataFrame(np.random.uniform(size=(5, 4)),
                      columns=['x', 'y', 'xe', 'ye'],
                      index=[1, 2, 3, 4, 5])
    plt.errorbar('x', 'y', xerr='xe', yerr='ye', data=df)

**Comment by gliptak** [comment]
Would this above be worked in a followup issue? Thanks

**Comment by diazona** [comment]
I think it's worth investigating how to do a more "proper" fix, but I don't have time to get on that right now. Perhaps it is better to just incorporate this commit into the master branch and take care of the other cases later with a separate issue. I guess it's up to the Pandas developers.

**Comment by jreback** [comment]
@gliptak if you can incorporate @diazona more extensive tests and make the changes I suggested.

You can debug the other case that was brought up and see if its possible to fix as well.

**Comment by tacaswell** [comment]
The pr into mpl fixes both issues.

**Comment by gliptak** [comment]
@jreback @diazona The code above directly calls into mpl, so changes to pandas will not offer a fix ...

File: pandas/tools/plotting.py (hunk 3 above)
Comment (gliptak) [inline pandas/tools/plotting.py]: updated

**Comment by gliptak** [comment]
Build fail seems to be unrelated:
```
  File "/home/travis/miniconda/envs/pandas/lib/python3.5/imp.py", line 234, in load_module

    return load_source(name, filename, file)

  File "/home/travis/miniconda/envs/pandas/lib/python3.5/imp.py", line 172, in load_source

    module = _load(spec)

  File "/usr/share/google/boto/boto_plugins/compute_auth.py", line 64

    except (urllib2.URLError, urllib2.HTTPError, IOError), e:

                                                         ^

SyntaxError: invalid syntax
```

**Comment by jreback** [comment]
yeah go ahead and rebase, fixed a weird boto import issue

**Comment by diazona** [comment]
I thought we were talking about allowing Pandas to work with older versions of matplotlib that don't have @tacaswell's patch. But if you'd rather say people should simply to upgrade matplotlib to avoid this bug, and leave it at that, I have no objections. Just checking that I'm on the same page as everyone else.

**Comment by jreback** [comment]
thanks @gliptak 

yeh important that pandas fixes this (and of course even moreso for mpl)

**Pull request closed without merging**